src/App.js

```diff
@@ -15,6 +15,16 @@ function App({ initialData }) {
     }
   })
 
+  const initialCategories = ['plant milk', 'plant yoghurt', 'meat alternative']
+
+  const [filterCategories, setFilterCategories] = useState(() => {
+    if (localStorage.getItem('categoriesLocalStorage')) {
+      return JSON.parse(localStorage.getItem('categoriesLocalStorage'))
+    } else {
+      return initialCategories
+    }
+  })
+
```

Comment:
Do you need all of this in the App.js? Might be better to put it in its file, to not loose overview.

Reply:
Will probably end up creating folders for the hooks and outsourcing them that way, I agree with your assessment, it's getting a bit crowded in the App.js 😸